File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        
        boolean answer = true;
        
        Deque<Character> que = new ArrayDeque<>();
        
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i) == ')' && !que.isEmpty() && que.peekLast() == '('){
                que.pollLast();
                continue;
            }
            que.add(s.charAt(i));
        }
        
        if(!que.isEmpty()) answer = false;

        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    
    var que = [];
    var answer = true;
    
    for(let i=0; i<s.length; i++){
        if(s.charAt(i) == ')' && que.length && que[que.length-1] == '('){
            que.pop();
            continue;
        }
        que.push(s.charAt(i));
    }
    
    if(que.length) answer = false;

    return answer;
}